### ISSUE TITLE
Lambda layer for custom local contrast normalization

### ISSUE THREAD

**Issue by matthewmok** [fchollet/keras#2918]
Hi,

I am trying to implement the local contrast normalization with a lambda layer in Keras.
 
<img width="345" alt="screen shot 2016-06-06 at 3 35 12 pm" src="https://cloud.githubusercontent.com/assets/10650116/15840416/a7ddaacc-2bfe-11e6-9fc8-f5e5eb2cdd92.png">
I am trying to create a custom "pooling" function and then create a lambda layer on top of it but having a hard time implementing it. The pseudo-code of the function is like this:
```
def lcn(x):
    current_pixel = (some pooling function that will return the center element of the window)
    mu = sum pooling (x)
    sigma = sqrt (sum pooling (x - mu))
    big_c = 1
    return (current_pixel - mu)/(sigma + big_c)
```
But I am not sure how to create a custom function that will return the center value of the window (for finding the current pixel) and creating a sum pooling (thought of using average pooling built in Keras but since it excludes padding and it is more complicated by simply multiplying the output of an average pooling by the size of the window). I tried to implemented a sum pooling using theano but I am not sure if it is correct or not. 
```
def sum_pool2d(x, strides = (7,7)):
    stride_win_size = strides[0] * strides[1]
    sum_x = T.pool.pool_2d(x, (1,1), st = strides, mode = 'average_inc_pad')
    sum_x *= stride_win_size
    return sum_x
```

Any help will be appreciated. 
Thanks!

**Comment by fchollet** [comment]
You could start from this legacy layer, shouldn't be difficult to adapt:
https://github.com/fchollet/keras/blob/97174dd298cf4b5be459e79b0181a124650d9148/keras/layers/normalization.py#L66

**Comment by kgrm** [comment]
What's the difference between "local contrast normalization" and batch normalization along the channel axis?

**Comment by matthewmok** [comment]
Thanks for the suggestion @fchollet . I ended up writing the normalization in a different way but this code should do the work:
```
# local contrast normalization on images
def lcn(x):
    ones_for_weight = np.reshape(np.ones((32, 32)), (1,32,32))
    mu = sum_pool2d(x, pool_size = (7, 7), strides = (1, 1), padding = (3, 3))
    mu_weight = sum_pool2d(ones_for_weight, pool_size = (7, 7), strides = (1, 1), padding = (3, 3))
    sum_sq_x = sum_pool2d(K.square(x), pool_size = (7, 7), strides = (1, 1), padding = (3, 3))
    total_mu_sq = mu_weight * K.square(mu)
    sq_cross_term = -2 * K.square(mu)
    sigma = K.sqrt(sum_sq_x + total_mu_sq + sq_cross_term)
    return (x - mu)/(sigma + 1)

def lcn_output_shape(input_shape):
    return input_shape


def sum_pool2d(x, pool_size = (7, 7), strides = (1, 1), padding = (3, 3)):
    sum_x = pool.pool_2d(x, ds = pool_size, st = strides, mode = 'sum', padding = padding, ignore_border = True)
    return sum_x

def sum_pool2d_output_shape(input_shape):
    return input_shape
```
